[[ Bug 17962 ]] Fix default object resetting

montegoulding opened the pull request:
This patch ensures that the defaultStack is valid
when resetting after a script may have deleted
the default stack.

This is an alternative and more complete fix to https://github.com/livecode/livecode/pull/4286

runrevmark commented on engine/src/object.h:
```diff
@@ -1491,7 +1491,7 @@ class MCObjectProxy::Handle
         return m_proxy != NULL && m_proxy->ObjectExists();
     }
-    operator bool() const
+    explicit operator bool() const
```
This change should probably be removed from this PR - I put it there when:

```
  if (t_object_handle)
   ...
```

Would work - even with it, it appears that the MCObject\* cast is used, and not the explicit bool. This is why I used the '.IsValid()' predicate in the ifs rather than relying on casts. @livecodefraser is back now so hopefully can work out why my attempts with 'explicit' failed!

montegoulding replied:
OK no problem @runrevmark

runrevmark commented:
Looks good - I notice you managed to ferret out a few more places with this pattern than I found!

@livecode-vulcan review ok 28ef13a

Commit:
[[ Bug 17962 ]] Fix default object resetting

This patch ensures that the defaultStack is valid
when resetting after a script may have deleted
the default stack.

montegoulding commented:
Sorry @runrevmark could you review again... just removed a whitespace diff from `object.h`

runrevmark commented:
@livecode-vulcan review ok 7bad821

livecode-vulcan commented:
💙 review by @runrevmark ok 7bad821b689174933f14d1a175b0f3e58c413e34

montegoulding commented:
Deleting because #4286 now includes this patch as they were merged to sort out the note filename clash
